megamek/src/megamek/client/bot/princess/BasicPathRanker.java

```diff
@@ -912,7 +921,9 @@ private double calcIceHazard(Entity movingUnit, Hex hex, MoveStep step, MovePath
                         breakthroughMod;
         logMsg.append("\n\t\tHazard value (")
               .append(LOG_DECIMAL.format(hazard)).append(").");
-        return hazard;
+        // Changed this to UNIT_DESTRUCTION_FACTOR because she suicided too often.
+        // No reason to be on the ice at all except as an absolute last resort.
+        return UNIT_DESTRUCTION_FACTOR;
```

Comment:
Be aware that returning a constant value here will essentially nullify the logic from line 915 through the end of the function.  The `hazard` return value can be 0 as assigned by `calcWaterHazard()` above it.  (i.e. for Amphibious types)

Consider maybe instead of returning a constant - maybe multiply `hazard` by some kind of "cryo-fear" factor - say 10 or something.  That way if `hazard` is 0, it will remain zero.

Reply:
Thank you.  Will look at tomorrow.

Reply:
Ok - so If I understand what you are saying - and I may not - yes this sets it to a fixed 1000 for everything remaining in this "might move on ice" group.  I did try to use incrementally larger numbers on ln 916, and no combination of numbers that I found prevented her from using the ice.  I also set a multiplier, and even increased those numbers to 999.5 and 999.1667. Nothing stopped her from using the ice until I set it to unit destruction factor.  (Although if you have another multiplier suggestion, I welcome it.  I was unable to code a working one).

I can move an exception for Aphibious units into this check, but that doesnt prevent the surface/breakage checks, and those are what are usually fatal, so I dont think even an amphibous vehicle should risk ice except as a last resort.

It would be different if this check prevented units from using water without ice, but it does not in my tests.  She moved mechs and hovers over and into water just fine even with this change.  (see screen below).

If I have misunderstood, or you have a better way to code it, I am happy to change.  I personally exhausted many non-working attempts before the final version I submitted, so improvement may be beyond my abilities.  Please let me what you would like for me to do.

![image](https://github.com/MegaMek/megamek/assets/23645569/eacc6e8d-1b74-4542-922e-a7b5a91c05b6)

Reply:
Talked to HS and answered question.  This is ready.